Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -352,9 +352,11 @@ namespace AzToolsFramework
         const AssetBrowserEntry* AssetBrowserTreeView::GetEntryByPath(QStringView path)
         {
             QModelIndex current;
-            for (auto token : AZStd::ranges::split_view(path, '/'))
+            bool currentlyFindingRows = false;
+            AZ::IO::Path p = path.toUtf8().constData();
+            for (auto it = p.begin(); it != p.end(); ++it)
             {
-                QStringView pathPart{ token.begin(), token.end() };
+                QString pathPart = QString::fromUtf8((*it).c_str());
                 const int rows = model()->rowCount(current);
                 bool rowFound = false;
                 for (int row=0; row < rows; ++row)
```

Comment:
If I'm reading this code correctly, the objective here is to find an item in the tree view's model based on the path parts in `path`, while skipping the leading parts of `path` that are not in the model. I think that using `QAbstractItemModel::match()` would be clearer than the loop made here. (I couldn't select the whole function, but I'd replace the whole body of the function with this version). This version also doesn't need the `currentlyFindingRows` bool, as that same information is encoded in the test `current.isValid()`.

Separately, this change results in two additional copies of the input `path`. The old version is using `split_view` along with `QStringView` to avoid any copies. With this version, the use of `split_view` is maintained, but each path part is copied into a `QString`, because the `match()` method works on a `QVariant` value, and `QVariant` doesn't work with `QStringView`.

```suggestion
            QModelIndex current;
            for (const auto token : AZStd::ranges::split_view(path, '/'))
            {
                QString pathPart(token.begin(), AZStd::distance(token.begin(), token.end()));

                QModelIndexList next = model()->match(
                    /*start =*/model()->index(0, 0, current),
                    /*role =*/Qt::DisplayRole,
                    /*value =*/pathPart,
                    /*hits =*/1,
                    /*flags =*/Qt::MatchExactly
                );
                if (next.size() == 1)
                {
                    current = next[0];
                }
                else if (current.isValid())
                {
                    return nullptr;
                }
            }
            return GetEntryFromIndex<AssetBrowserEntry>(current);
```